articles/quickstart/spa/auth0-react/01.md

```diff
@@ -0,0 +1,297 @@
+
+## Protecting a Route
+
+Your React application may implement navigation using a router. However, some routes may render public resources, while others may render private ones. In such a case, you can create an authentication route guard to render any component that defines a private resource conditionally. If Auth0 has authenticated the user, the guard renders the component; otherwise, the guard redirects the user to the [Auth0 Universal Login Page](https://auth0.com/docs/universal-login) to either log in or sign up.
+
+⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️
+
+**TODO: Final URL for additional routing examples.**
+ 
+⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️
+
+> React has no opinion on routing. This example assumes you are using [`react-router-dom`](https://github.com/ReactTraining/react-router#readme) &mdash; [access more examples for Gatsby and Next.js.](https://github.com/auth0-samples/auth0-react-sdk-sample)
+
+The default behavior is to redirect your users to the value of the `redirect_uri` prop of `Auth0Provider` after they log in or sign up using Auth0. However, you may want to redirect users directly to the private route they were trying to access. You can define such behavior using the `onRedirectCallback` prop of `Auth0Provider`.
+
+With React Router, your goal is to use the `history` object within the `onRedirectCallback` method to navigate to that private route programmatically. React Router exposes the `useHistory()` custom React Hook to make it easy to access the `history` object in any component that is a descendant of the `BrowserRouter` component.
+
+As such, you can create an `Auth0ProviderWithHistory` function component that renders `Auth0Provider`. This wrapper component then calls `useHistory()` within its body to make the `history` object available within the scope of `Auth0Provider` and its  `onRedirectCallback` event handler.
+
+**Root component:**
+
+```javascript
+ReactDOM.render(
+  <BrowserRouter>
+    <Auth0ProviderWithHistory>
+      <App />
+    </Auth0ProviderWithHistory>
+  </BrowserRouter>,
+  document.getElementById("root")
+);
+```
+
+**`Auth0ProviderWithHistory` component:**
+
+```javascript
+const Auth0ProviderWithHistory = ({ children }) => {
+  const history = useHistory();
+
+  const onRedirectCallback = (appState) => {
+    history.push(appState?.returnTo || window.location.pathname);
+  };
+
+  return (
+    <Auth0Provider
+      domain="${account.namespace}"
+      client_id="${account.clientId}"
+      redirect_uri={window.location.origin}
+      onRedirectCallback={onRedirectCallback}
+    >
+      {children}
+    </Auth0Provider>
+  );
+};
+```
+
+The implementation of the authentication route guard may look like this `PrivateRoute` component:
+
+```javascript
+const PrivateRoute = ({ component, ...args }) => (
+  <Route component={withLoginRequired(component)} {...args} />
```

Comment:
Where does withLoginRequired come from? No require statement.

Reply:
👍 I'll include the import.